openhtf/plugs/__init__.py

```diff
@@ -273,10 +272,14 @@ def __init__(self, plug_types=None, logger=None):
     self._xmlrpc_server = None
 
   def _asdict(self):
-    return {'plug_states': {name: plug._asdict()
-                            for name, plug in self._plugs_by_name.iteritems()},
-            'xmlrpc_port': self._xmlrpc_server and
-                           self._xmlrpc_server.socket.getsockname()[1]}
+    return {
+        'plug_descriptors': {name: self.get_plug_descriptor(plug)
```

Comment:
getting the plug descriptor can be expensive (even now, mro traversal isn't cached), do we want to cache it in a self._plug_descriptors?

Reply:
Hm, if plug descriptors don't change, then we can calculate plug_descriptors in `initialize_plugs()`.